test/test_file.cpp

```diff
@@ -408,3 +399,120 @@ TORRENT_TEST(stat_file)
 	TEST_CHECK(ec);
 	TEST_EQUAL(ec, boost::system::errc::no_such_file_or_directory);
 }
+
+// specificaly UNC tests
+#if TORRENT_USE_UNC_PATHS
+
+int maximum_component_length { 0 };
+bool support_hard_links { false };
+
+bool fill_current_directory_caps()
+{
+#ifdef TORRENT_WINDOWS
+	error_code ec;
+	DWORD dw_maximum_component_length;
+	DWORD dw_file_system_flags;
+	if (!GetVolumeInformationA(nullptr, nullptr, 0, nullptr, &dw_maximum_component_length, &dw_file_system_flags, nullptr, 0))
+	{
+		ec.assign(GetLastError(), system_category());
+		std::printf("GetVolumeInformation: [%s] %s\n"
+			, ec.category().name(), ec.message().c_str());
+		return false;
+	}
+	maximum_component_length = int(dw_maximum_component_length);
+	support_hard_links = ((dw_file_system_flags & FILE_SUPPORTS_HARD_LINKS) != 0);
```

Comment:
instead of writing to global variables, I think it would be a lot nicer if you would return a ``std::pair<bool, int>``. Then here you could say:

``return {support_hard_links, maximum_component_length};``

and in the test:

```
int maximum_component_length;
bool support_hard_links;
std::tie(support_hard_links, maximum_component_length) = current_directory_caps();
```

That way you also don't need any global variables